modules/artifact-activation/src/main/java/org/mule/runtime/module/artifact/activation/internal/maven/AbstractMavenDeployableProjectModelBuilder.java

```diff
@@ -107,6 +112,17 @@ protected AbstractMavenDeployableProjectModelBuilder(MavenConfiguration mavenCon
   public final DeployableProjectModel build() {
     File pom = getPomFromFolder(projectFolder);
 
+    Properties pomProperties;
+    Optional<String> version = empty();
+
+    if (projectFolder.equals(APP) || projectFolder.equals(DOMAIN) || projectFolder.isDirectory()) {
+      try {
+        pomProperties = getPomPropertiesFolder(projectFolder);
+        version = ofNullable(pomProperties.getProperty("version"));
+      } catch (Exception e) {
```

Comment:
handle this exception properly

Reply:
added a debug logger - there's a case where pom.properties doesn't exist and getPomPropertiesFolder would throw an exception for that. We are fine with not having the file as this is the first approach to find the artifact's version. just added a log.

Reply:
Shouldn't we catch FileNotFoundException or something similar here (just more specific than Exception)? just to avoid swallowing the exception in other cases

Reply:
replaced it with ArtifactDescriptorCreateException to specify the case